cli/server.go

```diff
@@ -327,7 +329,25 @@ func server() *cobra.Command {
 						return shutdownConnsCtx
 					},
 				}
-				errCh <- server.Serve(listener)
+
+				wg := errgroup.Group{}
+				wg.Go(func() error {
+					if dev && shouldTunnel {
+						defer tunnel.Listener.Close()
```

Comment:
I might add a comment here like: "The tunnel listener must be closed otherwise the wait group will be blocked forever!"